core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -71,12 +80,20 @@ public ChallengeResponse createChallenge(ChallengeRequest request) throws SepExc
     validateChallengeRequestClient(request);
     // Validate the validity of the memo
     Memo memo = validateChallengeRequestMemo(request);
+    String clientDomainSigningKey = null;
+    if (!isEmpty(request.getClientDomain())) {
+      debugF("Fetching SIGNING_KEY from client_domain: {}", request.getClientDomain());
+      clientDomainSigningKey = fetchSigningKeyFromClientDomain(request.getClientDomain());
+      debugF("SIGNING_KEY from client_domain fetched: {}", clientDomainSigningKey);
+    }
+    // Check authorization
+    validateAuthorization(request, authorization, clientDomainSigningKey);
```

Comment:
This is confusing because the signing key may not be the client domain signing key. 

The code does not clearly separate the case of the `client_domain` vs the `account` verification and can be confusing.

Reply:
It's always either client domain signing key or null, in which case does it seem to be not a client domain signing key?